assets/js/editor-components/compatibility-notice/index.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { Guide } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import { createInterpolateElement } from 'wordpress-element';
+import { useLocalStorageState } from '@woocommerce/base-hooks';
+import { isWpVersion } from '@woocommerce/settings';
+import type { ReactElement } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import WooImage from './woo-image';
+
+interface CompatibilityNoticeProps {
+	blockName: string;
+}
+
+export default function CompatibilityNotice( {
```

Comment:
I wonder if we should rename to CartCheckoutCompatibilityNotice as it is specific to these two. I think that if we need to add to other blocks in the future, we should change the logic to store only one key in localStorage `wc-blocks_compatibility_notice: ["cart","checkout","future_block"]` instead of keeping individual keys. But for now this works as expected.

Reply:
Great suggestions! In a1556e6ee7177078c60c3bbc6b0d13b6ef4c945c I renamed the component and in 06f2054afebf033e670bef9fc136a1e519876e6b I refactored it so it uses an array like the one you suggested.